cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -217,7 +219,15 @@ public static CruiseControl fromCrd(
             result.livenessProbeOptions = ProbeUtils.extractLivenessProbeOptionsOrDefault(ccSpec, ProbeUtils.DEFAULT_HEALTHCHECK_OPTIONS);
             result.gcLoggingEnabled = ccSpec.getJvmOptions() == null ? JvmOptions.DEFAULT_GC_LOGGING_ENABLED : ccSpec.getJvmOptions().isGcLoggingEnabled();
             result.jvmOptions = ccSpec.getJvmOptions();
-            result.metrics = new MetricsModel(ccSpec);
+
+            /*
+            Metrics Reporter is not yet supported with CruiseControl as CC's own metrics are only exposed through JMX.
+            Metrics config type check happens with CEL validation at the spec level.
+            */
+            if (ccSpec.getMetricsConfig() instanceof JmxPrometheusExporterMetrics) {
+                result.metrics = new JmxPrometheusExporterModel(ccSpec);
+            }
```

Comment:
You should probably add and `else` clause and thrown `InvalidResourceRException` or something.

Reply:
@scholzj Thanks for the review.
We have CEL validation for this so we should not need the exception right??. Although I see in a later comment that we have the else clause in KafkaConnect and MM2. Should we remove the else clause and have CEL validation for all three of them?